docs/source/tutorials/vanilla-js.md

```diff
@@ -0,0 +1,281 @@
+
+In this tutorial, you’ll build a simple JavaScript application that detects objects in images using Transformers.js. The app requires no server, external libraries, or build tools... all you need is a code editor and a browser!
+
+Here's how it works: the user uploads an image to the browser via the “Upload image” button. The app then downloads an AI model, runs the image through it, and finally displays an overview of where the different objects are located, like this:
+
+![Demo](https://huggingface.co/datasets/Xenova/transformers.js-docs/resolve/main/js-detection-interence-zebra.png)
+
+**************************Useful links:**************************
+
+- [Demo site](https://huggingface.co/spaces/Scrimba/javascript-object-detector)
+- [Interactive code walk-through (scrim)](https://scrimba.com/scrim/cKm9bDAg)
+- [Source code](https://huggingface.co/spaces/Scrimba/javascript-object-detector/tree/main)
+
+## Step 1:  HTML and CSS setup
+
+The first thing we'll do is create some markup and styling. Create an in `index.html` file with a basic HTML skeleton, and add the following `<main>` tag to the `<body>`:
+
+```html
+<main class="container">
+  <label class="custom-file-upload">
+    <input id="file-upload" type="file" accept="image/*" />
+    <img class="upload-icon" src="https://huggingface.co/datasets/Xenova/transformers.js-docs/resolve/main/upload-icon.png" />
+    Upload image
+  </label>
+  <div id="image-container"></div>
+  <p id="status"></p>
+</main>
+```
+
+Let’s break it down:
+
+We’re adding an `<input>` element with `type="file"`. This gives the user the ability to pick an image from their local files. The default styling for this element looks pretty bad, so we are going to style it as a nice button instead. The easiest way to achieve this is to wrap the `<input>` element in a `<label>`, hide the input, and then style the label as a button.
+
+We’re also adding an empty `<div>` container for displaying the image, plus an empty `<p>` tag that we'll use to give status updates to the user as we're downloading and running the model, as both of these operations take some time.
+
+Next, add the following CSS rules in a `style.css` file and and link it to the HTML:
+
+```css
+html,
+body {
+    font-family: Arial, Helvetica, sans-serif;
+}
+
+.container {
+    margin: 40px auto;
+    width: max(50vw, 400px);
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+}
+
+.custom-file-upload {
+    display: flex;
+    align-items: center;
+    cursor: pointer;
+    gap: 10px;
+    border: 2px solid black;
+    padding: 8px 16px;
+    cursor: pointer;
+    border-radius: 6px;
+}
+
+#file-upload {
+    display: none;
+}
+
+.upload-icon {
+    width: 30px;
+}
+
+#image-container {
+    width: 100%;
+    margin-top: 20px;
+    position: relative;
+}
+
+#image-container>img {
+    width: 100%;
+}
+```
+
+If you drag and drop the `index.html` file into a browser, you should see the following:
```

Comment:
I think that adding this line might be a bit confusing later on, because when they refresh it will show a [CORS](https://stackoverflow.com/questions/52919331/access-to-script-at-from-origin-null-has-been-blocked-by-cors-policy) error after they try adding JS.

> Access to script at 'file:///.../index.js' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, isolated-app, chrome-extension, chrome, https, chrome-untrusted.

Ideally we could link to some online playground for this. Of course, using a scrim would be great, but it requires a sign up before starting to use the editor (unlike jsfiddle, codepen, or codesandbox). WDYT?

Reply:
Ah, got it. You actually don't need an account to start editing a scrim, so we can point to a scrim and let them run the code. Does that sound good?

Reply:
Sure that works! 😇 In that case, we can just link to the ready-made scrim (unless you think it's better to link to an empty one?)

Reply:
How about I link to a scrim which mirrors the code at that exact point in the tutorial?

Reply:
Is there a way to have "chapters" or "versions" of a scrim? In that way, we can link to a version of the scrim that is not yet complete, but the user can click a button or continue the video/walkthrough to get the rest of it?

Reply:
We can link to specific times in the timeline, so that they can continue watching from that point. For example, here's a link to the `runModel() ` explanation of the old scrim: https://scrimba.com/scrim/co87f4ae7aea87a94a732f298?a=7877.2.3.L2_31-2_45

Reply:
I updated the text to simply not talk about them running the code themselves. That way we dodge the whole CORS issue in that sentence.

Have added a mention about the need for a server in the first paragraph though.

Reply:
Sounds good! 👍